game-app/game-core/src/test/java/games/strategy/triplea/delegate/data/CasualtyDetailsTest.java

```diff
@@ -94,8 +94,9 @@ void killLowestMovementAirUnitsWhenOnlyOneTypeIsAvailable() {
   }
 
   @Test
-  void damageLowestMovementAirUnitsWhenOnlyOneTypeIsAvailable() {
+  void damageHighestMovementAirUnitsWhenOnlyOneTypeIsAvailable() {
     final UnitType fighter = givenUnitType("fighter");
+    UnitAttachment.get(fighter).setHitPoints(2);
```

Comment:
If all things are equal, we want an air unit with lowest movement to be taken as a casualty first.
I think this means we should keep the existing test to ensure that behavior continues.

A new unit test would be good here though to verify that we prefer units with multiple hit points before units with less movement.

Reply:
If all things are equal, we want an air unit with lowest movement to be _killed_ first.
I want to move a damaged air unit as far as possible away from the battle field. So I want to allocate damage to an air unit with _highest_ movement first. I can imagine rare cases where you would want to allocate damage to an air unit with the lowest movement first, but they seem very rare to me. Can you give an example?

> A new unit test would be good here though to verify that we prefer units with multiple hit points before units with less movement.

I don't understand. Do you mean we take damage without killing a unit before we kill a unit? If so: I agree, and that is the implemented functionality. Would you like me to write a test case for that?

Reply:
There was a recent bug report where arbitrary air units were being taken as casualties. This is sub-optimal if air units with less movement points are taken first. I think you mentioned somewhere that damage should be to units with the most movement, and then casualties taken from those with the least.

Reply:
The existing unit tests **killLowestMovementAirUnitsWhenOnlyOneTypeIsAvailable** and **killLowestMovementAirUnitsInTwoTypes** take care for this.